src/logging_layer.rs

```diff
@@ -17,4 +17,31 @@ pub fn initialize_logging_layer() {
     if builder.try_init().is_ok() {
         log::info!("Initialized the logger ..")
     }
+
+    log_detected_x86_features();
 }
+
+fn log_detected_x86_features() {
+    let mut features: Vec<String> = Vec::new();
+    if is_x86_feature_detected!("avx") {
+        features.push("AVX".to_string());
+    }
+
+    if is_x86_feature_detected!("avx2") {
+        features.push("AVX2".to_string());
+    }
+
+    if is_x86_feature_detected!("avx512f") {
+        features.push("AVX512F".to_string());
+    }
+
+    if is_x86_feature_detected!("fma") {
+        features.push("FMA".to_string());
+    }
+
+    if features.is_empty() {
+        log::info!("No CPU features detected ..");
```

Comment:
This should be "No _selected_ CPU features detected" imo, otherwise one would think we're running on air here

Reply:
Updated